src/app/page.tsx

```diff
@@ -1,41 +1,158 @@
 'use client'
 
-import { Box } from '@primer/react'
-import { useSession } from 'next-auth/react'
-import { useRouter } from 'next/navigation'
-import { useEffect, useState } from 'react'
-import { personalOctokit } from '../bot/octokit'
-import { Login } from './components/login/Login'
+import { Avatar, Box, Link, Octicon } from '@primer/react'
+import { use, useEffect, useState } from 'react'
+import { OrgsData, useOrgsData } from 'hooks/useOrganizations'
+import { Search } from './components/search/Search'
+import { DataTable, Table } from '@primer/react/lib-esm/DataTable'
+import Fuse from 'fuse.js'
+import Blankslate from '@primer/react/lib-esm/Blankslate/Blankslate'
+import { OrganizationIcon } from '@primer/octicons-react'
+import { Stack } from '@primer/react/lib-esm/Stack'
+import { WelcomeHeader } from './components/header/WelcomeHeader'
 
 const Home = () => {
-  const router = useRouter()
-  const session = useSession()
-  const [organizations, setOrganizations] = useState<
-    Awaited<ReturnType<typeof getAllOrganizations>>
-  >([])
-
-  const getAllOrganizations = async (accessToken: string) => {
-    const octokit = personalOctokit(accessToken)
-    const data = await octokit.rest.orgs.listForAuthenticatedUser()
-    return data.data
+  const orgsData = useOrgsData()
+
+  // set search value to be empty string by default
+  const [searchValue, setSearchValue] = useState('')
+
+  // values for pagination
+  const pageSize = 5
+  const [pageIndex, setPageIndex] = useState(0)
+  const start = pageIndex * pageSize
+  const end = start + pageSize
+
+  // show loading table
+  if (!orgsData) {
+    return (
+      <Box>
+        <WelcomeHeader />
+        <Search
+          placeholder="Find an organization"
+          searchValue={searchValue}
+          setSearchValue={setSearchValue}
+        />
+        <Table.Container>
+          <Table.Skeleton
+            columns={[
+              {
+                header: 'Organization',
+                rowHeader: true,
+              },
+            ]}
+            rows={5}
+            cellPadding="spacious"
+          />
+          <Table.Pagination aria-label="pagination" totalCount={0} />
+        </Table.Container>
+      </Box>
+    )
   }
 
-  useEffect(() => {
-    if (session.data?.user) {
-      const { accessToken } = session.data.user ?? {}
-      if (!accessToken) {
-        return
-      }
+  let orgs = orgsData
```

Comment:
If you meant to make a copy here, you'll need to perform a spread

```suggestion
  let orgs = [...orgsData]
```

TS/JS is reference based for anything that isn't a primitive so that means you'll be modifying the underlying array

```ts
const data = [1,2,3]
let data2 = data
data2.splice(0,1)
console.log(data, data2)
// (2) [2, 3] (2) [2, 3]
```

Same thing but with spread operator

```ts
const data = [1,2,3]
let data2 = [...data]
data2.splice(0,1)
console.log(data, data2)
// (3) [1, 2, 3] (2) [2, 3]
```